packages/hardhat-core/src/internal/hardhat-network/provider/vm/ethereumjs.ts

```diff
@@ -764,48 +765,61 @@ export class EthereumJSAdapter implements VMAdapter {
             : SuccessReason.Stop;
 
         executionResult = {
-          reason,
-          gasUsed,
-          gasRefunded: result.execResult.gasRefund ?? 0n,
-          logs:
-            result.execResult.logs?.map((log) => {
-              return {
-                address: log[0],
-                topics: log[1],
-                data: log[2],
-              };
-            }) ?? [],
-          output:
-            result.createdAddress === undefined
-              ? {
-                  returnValue: result.execResult.returnValue,
-                }
-              : {
-                  address: result.createdAddress.toBuffer(),
-                  returnValue: result.execResult.returnValue,
-                },
+          result: {
+            reason,
+            gasUsed,
+            gasRefunded: result.execResult.gasRefund ?? 0n,
+            logs:
+              result.execResult.logs?.map((log) => {
+                return {
+                  address: log[0],
+                  topics: log[1],
+                  data: log[2],
+                };
+              }) ?? [],
+            output:
+              result.createdAddress === undefined
+                ? {
+                    returnValue: result.execResult.returnValue,
+                  }
+                : {
+                    address: result.createdAddress.toBuffer(),
+                    returnValue: result.execResult.returnValue,
+                  },
+          },
         };
       } else if (result.execResult.exceptionError.error === ERROR.REVERT) {
         executionResult = {
-          gasUsed,
-          output: result.execResult.returnValue,
+          result: {
+            gasUsed,
+            output: result.execResult.returnValue,
+          },
         };
       } else {
-        const vmError = Exit.fromEthereumJSEvmError(
+        // Override the exceptional halt reason with the one from the EVM result, to avoid lossy casting
```

Comment:
I'm not sure I understand this. The idea is to avoid converting it twice? Is that the lossy part?